api/v1alpha1/storageconsumer_types.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2021 Red Hat OpenShift Container Storage.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StorageConsumerState represent a StorageConsumer's state
+type StorageConsumerState string
+
+const (
+	// StateReady represents Ready state of an object
+	StateReady StorageConsumerState = "Ready"
+	// StateConfiguring represents Configuring state of an object
+	StateConfiguring StorageConsumerState = "Configuring"
```

Comment:
Maybe `Progressing` instead of `Configuring`?

Reply:
We are not defining a phase here but a state, on top of that I think progressing is too generic

Reply:
Another point is that the status section of a CR should not reflect the operator's status but the CR status. In ocs operator we kind of abused the fact that we only supported a single instance of the stroagecluster CR and reported the operator status (like progressing for phase). 

I do not want to repeat the same mistakes again.